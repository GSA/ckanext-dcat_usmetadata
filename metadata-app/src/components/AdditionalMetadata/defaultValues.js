export default {
  dataQualityUSG: '',
  themes: '',
  describedBy: '',
  describedByType: '',
  accrualPeriodicity: '',
  landingPage: '',
  language: '',
  primaryITInvestmentUIIUSG: '',
  references: '',
  issued: '',
  systemOfRecordsUSG: '',
  isPartOf: '',
};
