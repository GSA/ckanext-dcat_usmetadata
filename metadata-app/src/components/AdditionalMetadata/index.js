import React from 'react';
import PropTypes from 'prop-types';
import WrappedField from '../WrappedField';
import HelpText from '../HelpText';

const helpTexts = {
  themes: <HelpText>Examples include: vegetables, non_starchy, green.</HelpText>,
  describedBy: (
    <HelpText>
      Provide a link to data dictionary or other reference that helps users understand the dataset.
      Example: &quot;http://www.agency.gov/vegetables/vegetables-all.zip&quot;
    </HelpText>
  ),
  landingPage: (
    <HelpText>
      This landing page should contain or be linked to all resources tied to the dataset. Example:
      &quot;http://www.agency.gov/vegetables&quot;
    </HelpText>
  ),
  primaryITInvestmentUIIUSG: (
    <HelpText>
      This can often be found in Exhibit 53 documents, and takes the form ###-#########. Learn more
      about Exhibit 53 documents at{' '}
      <a href="https://www.whitehouse.gov/wp-content/uploads/2018/06/fy-2020-it-budget-guidance.pdf">
        this pdf
      </a>
    </HelpText>
  ),
  references: (
    <HelpText>
      Related documents such as technical information about a dataset, developer documentation, etc.
      Format for entry is: http://www.agency.gov/fruits/fruits.csv,
      http://www.agency.gov/legumes/legumes
    </HelpText>
  ),
  systemOfRecordsUSG: (
    <HelpText>
      If the dataset is a designated System of Records under the Privacy Act of 1974, provide the
      URL for the System of Recors Notice
    </HelpText>
  ),
};

const AdditionalMetadata = (props) => {
  const { values } = props;
  const formValues = values || {};
  const {
    dataQualityUSG,
    themes,
    describedBy,
    describedByType,
    accrualPeriodicity,
    landingPage,
    language,
    primaryITInvestmentUIIUSG,
    references,
    issued,
    systemOfRecordsUSG,
    isPartOf,
  } = formValues;

  return (
    <div className="usa-form-custom">
      <section id="section-basic-mega-menu" className="site-component-section">
        <p className="site-text-intro">
          Please note that the additional metadata that you upload will help public users better
          find and use this dataset. Not all of these criteria will apply to each dataset, so feel
          free to only answer what applies. For more information about the form fields, consult the{' '}
          <a href="https://resources.data.gov/schemas/dcat-us/v1.1/">DCAT-US Schema</a>
        </p>
      </section>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Meets Agency Data Quality"
            name="dataQualityUSG"
            type="select"
            choices={['Yes', 'No']}
            value={dataQualityUSG}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Themes"
            name="themes"
            type="string"
            helptext="Main thematic category of the dataset.  If this dataset should be included in geoplatform.gov, please enter “geospatial” as the theme.  Start typing to add themes."
            infoText={helpTexts.themes}
            value={themes}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Data Dictionary URL"
            name="describedBy"
            type="string"
            helptext='Please include "http://" at the beginning of your URL.'
            infoText={helpTexts.describedBy}
            value={describedBy}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Data Dictionary Type"
            name="describedByType"
            type="string"
            value={describedByType}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Data Publishing Frequency"
            name="accrualPeriodicity"
            type="string"
            value={accrualPeriodicity}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Dataset Landing Page URL"
            name="landingPage"
            type="string"
            helptext='Please include "http://" at the beginning of your URL.'
            infoText={helpTexts.landingPage}
            value={landingPage}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Language - Language Subtag"
            name="language"
            type="string"
            value={language}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Dataset's IT Unique Investment Identifier"
            name="primaryITInvestmentUIIUSG"
            type="string"
            value={primaryITInvestmentUIIUSG}
            infoText={helpTexts.primaryITInvestmentUIIUSG}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Related Documents"
            name="references"
            type="string"
            value={references}
            infoText={helpTexts.references}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Release Date"
            name="issued"
            type="date"
            value={issued}
            infoText={helpTexts.references}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Release Date"
            name="systemOfRecordsUSG"
            type="string"
            value={systemOfRecordsUSG}
            infoText={helpTexts.systemOfRecordsUSG}
          />
        </div>
      </div>
      <div className="row">
        <div className="grid-col-12">
          <WrappedField
            label="Select Parent Dataset"
            name="isPartOf"
            type="string"
            value={isPartOf}
          />
        </div>
      </div>
      <div className="row">
        <button type="button" className="usa-button usa-button--line">
          Save as draft
        </button>
        <button className="usa-button" type="submit">
          Save and continue
        </button>
      </div>
    </div>
  );
};

AdditionalMetadata.propTypes = {
  values: PropTypes.shape({
    dataQualityUSG: PropTypes.string,
    themes: PropTypes.string,
    describedBy: PropTypes.string,
    describedByType: PropTypes.string,
    accrualPeriodicity: PropTypes.string,
    landingPage: PropTypes.string,
    language: PropTypes.string,
    primaryITInvestmentUIIUSG: PropTypes.string,
    references: PropTypes.string,
    issued: PropTypes.string,
    systemOfRecordsUSG: PropTypes.string,
    isPartOf: PropTypes.string,
  }),
};

export default AdditionalMetadata;
