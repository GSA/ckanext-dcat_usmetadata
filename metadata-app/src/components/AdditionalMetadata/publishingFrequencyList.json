[
  {
    "value": "irregular",
    "label": "Irregular/As needed"
  },
  {
    "value": "Continuously updated",
    "label": "Continuously"
  },
  {
    "value": "R/P1D",
    "label": "Daily"
  },
  {
    "value": "R/P1W",
    "label": "Weekly"
  },
  {
    "value": "R/P1M",
    "label": "Monthly"
  },
  {
    "value": "R/P3M",
    "label": "Quarterly"
  },
  {
    "value": "R/P1Y",
    "label": "Yearly"
  },
  {
    "value": "other",
    "label": "Other"
  }
]
