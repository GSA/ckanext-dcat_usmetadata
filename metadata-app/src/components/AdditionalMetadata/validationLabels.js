export default {
  title: 'Title',
  contactPoint: 'Contact Name',
  identifier: 'Unique ID',
  contactEmail: 'Contact Email',
  description: 'Description',
  publisher: 'Publisher',
  accessLevel: 'Public Access Level',
  dataQuality: 'Data Quality',
  rights: 'Rights',
  spatial: 'Spatial',
  license: 'License',
  temporal: 'Temporal',
  release_date: 'Release Date',
  accrualPeriodicityOther: 'Data Publishing Frequency',
};
