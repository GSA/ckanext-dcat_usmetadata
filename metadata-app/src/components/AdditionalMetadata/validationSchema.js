import * as yup from 'yup';

export default yup.object().shape({
  dataQualityUSG: yup.string(),
  themes: yup.string(),
  describedBy: yup.string(),
  describedByType: yup.string(),
  accrualPeriodicity: yup.string(),
  landingPage: yup.string(),
  language: yup.string(),
  primaryITInvestmentUIIUSG: yup.string(),
  references: yup.string(),
  issued: yup.string(),
  systemOfRecordsUSG: yup.string(),
  isPartOf: yup.string(),
});
