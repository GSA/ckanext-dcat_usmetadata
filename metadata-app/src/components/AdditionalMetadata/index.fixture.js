import React from 'react';
import { Formik } from 'formik';
import AdditionalMetadata from '.';
import '../../css/custom.css';

const apiUrl = 'http://localhost:5000/api/3/action/';
const apiKey = '6556b77b-b96b-4352-999c-bde4a0d86cce';

const Form = () => (
  <div className="grid-container">
    <Formik
      initialValues={{}}
      render={(values) => (
        <form>
          <AdditionalMetadata
            apiKey={apiKey}
            apiUrl={apiUrl}
            currentStep={1}
            fetchDatasetsOpts="false"
            values={values}
          />
        </form>
      )}
    />
  </div>
);

export default <Form />;
