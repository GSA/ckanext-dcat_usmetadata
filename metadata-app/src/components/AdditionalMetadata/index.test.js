import React from 'react';
import { Formik } from 'formik';
import { render } from '@testing-library/react';
import AdditionalMetadata from '.';
import '../../css/custom.css';

const Form = () => (
  <Formik initialValues={{}}>
    {(values) => (
      <form>
        <AdditionalMetadata
          apiUrl="http://localhost:5000/api/3/action"
          apiKey="123"
          currentStep={2}
          fetchDatasetsOpts="false"
          values={values}
        />
      </form>
    )}
  </Formik>
);

test('Renders RequiredMetadata component', () => {
  const { getByText } = render(<Form />);
  const text1 = getByText('Meets Agency Data Quality');
  const text2 = getByText('Data Dictionary Type');
  const text3 = getByText('Select Parent Dataset');
  expect(text1).toBeInTheDocument();
  expect(text2).toBeInTheDocument();
  expect(text3).toBeInTheDocument();
});
