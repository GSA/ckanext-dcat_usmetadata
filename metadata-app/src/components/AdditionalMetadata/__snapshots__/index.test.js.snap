// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RequiredMetadata component 1`] = `
<div>
  <form>
    <div
      class="usa-form-custom"
    >
      <section
        class="site-component-section"
        id="section-basic-mega-menu"
      >
        <p
          class="site-text-intro"
        >
          Please note that the additional metadata that you upload will help public users better find and use this dataset. Not all of these criteria will apply to each dataset, so feel free to only answer what applies. For more information about the form fields, consult the
           
          <a
            class="usa-link usa-link--external"
            href="https://resources.data.gov/schemas/dcat-us/v1.1/"
            rel="noopener noreferrer"
            target="_blank"
          >
            DCAT-US Schema
          </a>
          .
        </p>
      </section>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Meets Agency Data Quality
            </label>
            <div
              class="usa-helptext "
            />
            
            <select
              choices="Yes,No"
              class="usa-select  "
              name="dataQuality"
              type="select"
            >
              <option
                value=""
              >
                -Select-
              </option>
              <option
                value="Yes"
              >
                Yes
              </option>
              <option
                value="No"
              >
                No
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <span
            class="usa-label"
          >
            Geospatial
          </span>
          <div
            class="usa-helptext"
          >
            Geospatial datasets are included in
             
            <a
              class="usa-link usa-link--external"
              href="https://www.geoplatform.gov/"
              rel="noopener noreferrer"
              target="_blank"
            >
              geoplatform.gov
            </a>
            .
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="category-option-yes"
              name="category"
              type="radio"
              value="geospatial"
            />
            <label
              class="usa-radio__label"
              for="category-option-yes"
            >
              My dataset is a geospatial dataset
            </label>
          </div>
          <div
            class="form-group "
          >
            <input
              class="usa-radio__input"
              id="category-option-no"
              name="category"
              type="radio"
              value=""
            />
            <label
              class="usa-radio__label"
              for="category-option-no"
            >
              My dataset is not a geospatial dataset
            </label>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Data Dictionary URL
              <div
                class="tooltip"
              >
                <svg
                  height="20px"
                  style="margin-left: .5em;"
                  tabindex="0"
                  width="20px"
                >
                  info.svg
                </svg>
              </div>
            </label>
            <div
              class="usa-helptext "
            >
              Please include "https://" at the beginning of your URL.
            </div>
            
            <input
              class="usa-input  undefined undefined"
              name="data_dictionary"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Data Dictionary Type
            </label>
            <div
              class="usa-helptext "
            />
            
            <select
              choices="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
              class="usa-select  "
              name="describedByType"
              type="select"
            >
              <option
                value=""
              >
                -Select-
              </option>
              <option
                value="ArcGIS Map Preview"
              >
                ArcGIS Map Preview - ArcGIS Map Preview
              </option>
              <option
                value="ArcGIS Map Service"
              >
                ArcGIS Map Service - ArcGIS Map Service
              </option>
              <option
                value="ArcGIS Online Map"
              >
                ArcGIS Online Map - ArcGIS Online Map
              </option>
              <option
                value="application/atom+xml"
              >
                Atom Feed - Atom Feed
              </option>
              <option
                value="application/octet-stream"
              >
                BIN - Binary Data
              </option>
              <option
                value="image/x-ms-bmp"
              >
                BMP - Bitmap Image File
              </option>
              <option
                value="text/csv"
              >
                CSV - Comma Separated Values File
              </option>
              <option
                value="application/x-dbf"
              >
                dBase - dBase Database
              </option>
              <option
                value="application/x-director"
              >
                DCR - Adobe Shockwave format
              </option>
              <option
                value="application/msword"
              >
                DOC - Word Document
              </option>
              <option
                value="application/vnd.openxmlformats-officedocument.wordprocessingml.document"
              >
                DOCX - Word OOXML Document
              </option>
              <option
                value="application/x-e00"
              >
                E00 -  ARC/INFO interchange file format
              </option>
              <option
                value="Esri REST"
              >
                Esri REST - Esri Rest API Endpoint
              </option>
              <option
                value="application/x-msdownload"
              >
                EXE - Windows Executable Program
              </option>
              <option
                value="[object Object]"
              >
                GeoJSON - Geographic JavaScript Object Notation
              </option>
              <option
                value="image/gif"
              >
                GIF - GIF Image File
              </option>
              <option
                value="application/gzip"
              >
                GZ - Gzip File
              </option>
              <option
                value="text/html"
              >
                HTML - Web Page
              </option>
              <option
                value="image/jpeg"
              >
                JPEG - JPG Image File
              </option>
              <option
                value="application/x-javascript"
              >
                JS - JavaScript
              </option>
              <option
                value="application/json"
              >
                JSON - JavaScript Object Notation
              </option>
              <option
                value="application/vnd.google-earth.kml+xml"
              >
                KML - KML File
              </option>
              <option
                value="application/vnd.google-earth.kmz+xml"
              >
                KMZ - KMZ File
              </option>
              <option
                value="application/x-msaccess"
              >
                MDB - Access Database
              </option>
              <option
                value="chemical/x-mopac-input"
              >
                MOP - MOPAC Input format
              </option>
              <option
                value="image/x-mrsid"
              >
                MrSID - MrSID
              </option>
              <option
                value="application/x-mxd"
              >
                MXD - ESRI ArcGIS project file
              </option>
              <option
                value="application/x-n3"
              >
                N3 - N3 Triples
              </option>
              <option
                value="application/netcdf"
              >
                NetCDF - NetCDF File
              </option>
              <option
                value="application/vnd.oasis.opendocument.database"
              >
                ODB - OpenDocument Database
              </option>
              <option
                value="application/vnd.oasis.opendocument.chart"
              >
                ODC - OpenDocument Chart
              </option>
              <option
                value="application/vnd.oasis.opendocument.formula"
              >
                ODF - OpenDocument Math Formula
              </option>
              <option
                value="application/vnd.oasis.opendocument.graphics"
              >
                ODG - OpenDocument Image
              </option>
              <option
                value="application/vnd.oasis.opendocument.presentation"
              >
                ODP - OpenDocument Presentation
              </option>
              <option
                value="application/vnd.oasis.opendocument.spreadsheet"
              >
                ODS - OpenDocument Spreadsheet
              </option>
              <option
                value="application/vnd.oasis.opendocument.text"
              >
                ODT - OpenDocument Text
              </option>
              <option
                value="other"
              >
                Other
              </option>
              <option
                value="application/owl+xml"
              >
                OWL - Web Ontology Language
              </option>
              <option
                value="application/pdf"
              >
                PDF - PDF File
              </option>
              <option
                value="text/x-perl"
              >
                Perl - Perl Script
              </option>
              <option
                value="image/png"
              >
                PNG - PNG Image File
              </option>
              <option
                value="application/vnd.ms-powerpoint"
              >
                PPT - Powerpoint Presentation
              </option>
              <option
                value="application/vnd.openxmlformats-officedocument.presentationml.presentation"
              >
                PPTX - Powerpoint OOXML Presentation
              </option>
              <option
                value="application/x-qgis"
              >
                QGIS - QGIS File
              </option>
              <option
                value="application/rar"
              >
                RAR - RAR Compressed File
              </option>
              <option
                value="application/rdf+xml"
              >
                RDF - RDF
              </option>
              <option
                value="application/rss+xml"
              >
                RSS - RSS feed
              </option>
              <option
                value="[object Object]"
              >
                SHP - Shapefile
              </option>
              <option
                value="application/sparql-results+xml"
              >
                SPARQL - SPARQL end-point
              </option>
              <option
                value="image/svg+xml"
              >
                SVG - SVG vector image
              </option>
              <option
                value="application/x-tar"
              >
                TAR - TAR Compressed File
              </option>
              <option
                value="image/tiff"
              >
                TIFF - TIFF Image File
              </option>
              <option
                value="text/tab-separated-values"
              >
                TSV - Tab Separated Values File
              </option>
              <option
                value="text/plain"
              >
                TXT - Text File
              </option>
              <option
                value="wcs"
              >
                WCS - Web Coverage Service
              </option>
              <option
                value="[object Object]"
              >
                WFS - Web Feature Service
              </option>
              <option
                value="WMS"
              >
                WMS - Web Mapping Service
              </option>
              <option
                value="application/vnd.ms-excel"
              >
                XLS - Excel Document
              </option>
              <option
                value="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
              >
                XLSX - Excel OOXML Spreadsheet
              </option>
              <option
                value="application/xml"
              >
                XML - XML File
              </option>
              <option
                value="application/xslt+xml"
              >
                XSLT - Extensible Stylesheet Language Transformations
              </option>
              <option
                value="chemical/x-xyz"
              >
                XYZ - XYZ Chemical File
              </option>
              <option
                value="application/zip"
              >
                ZIP - Zip File
              </option>
            </select>
          </div>
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            />
            <div
              class="usa-helptext disabled"
            >
              If you selected “Other”, please specify your Data Dictionary Type
            </div>
            
            <input
              class="usa-input disabled undefined undefined"
              disabled=""
              name="otherDataDictionaryType"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Data Publishing Frequency
            </label>
            <div
              class="usa-helptext "
            />
            
            <select
              choices="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
              class="usa-select  "
              name="accrualPeriodicity"
              type="select"
            >
              <option
                value=""
              >
                -Select-
              </option>
              <option
                value="irregular"
              >
                Irregular/As needed
              </option>
              <option
                value="Continuously updated"
              >
                Continuously
              </option>
              <option
                value="R/P1D"
              >
                Daily
              </option>
              <option
                value="R/P1W"
              >
                Weekly
              </option>
              <option
                value="R/P1M"
              >
                Monthly
              </option>
              <option
                value="R/P3M"
              >
                Quarterly
              </option>
              <option
                value="R/P1Y"
              >
                Yearly
              </option>
              <option
                value="other"
              >
                Other
              </option>
            </select>
          </div>
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            />
            <div
              class="usa-helptext disabled"
            >
              <span
                class="usa-helptext"
              >
                If you selected “Other, please specify a valid
                 
                <a
                  class="usa-link usa-link--external"
                  href="https://en.wikipedia.org/wiki/ISO_8601"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  ISO 8601
                </a>
                 
                timestamp
              </span>
            </div>
            
            <input
              class="usa-input disabled undefined undefined"
              disabled=""
              name="accrualPeriodicityOther"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Dataset Landing Page URL
              <div
                class="tooltip"
              >
                <svg
                  height="20px"
                  style="margin-left: .5em;"
                  tabindex="0"
                  width="20px"
                >
                  info.svg
                </svg>
              </div>
            </label>
            <div
              class="usa-helptext "
            >
              Please include "https://" at the beginning of your URL.
            </div>
            
            <input
              class="usa-input  undefined undefined"
              name="homepage_url"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Language - Language Subtag
            </label>
            <div
              class="usa-helptext "
            />
            
            <select
              choices="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
              class="usa-select  "
              name="languageSubTag"
              type="select"
            >
              <option
                value=""
              >
                -Select-
              </option>
              <option
                value="af"
              >
                Afrikaans 
              </option>
              <option
                value="sq"
              >
                Albanian 
              </option>
              <option
                value="ar"
              >
                Arabic 
              </option>
              <option
                value="hy"
              >
                Armenian 
              </option>
              <option
                value="Cy"
              >
                Uzbek (Cyrillic) 
              </option>
              <option
                value="Lt"
              >
                Uzbek (Latin) 
              </option>
              <option
                value="eu"
              >
                Basque 
              </option>
              <option
                value="be"
              >
                Belarusian 
              </option>
              <option
                value="bg"
              >
                Bulgarian 
              </option>
              <option
                value="ca"
              >
                Catalan 
              </option>
              <option
                value="zh"
              >
                Chinese
              </option>
              <option
                value="hr"
              >
                Croatian 
              </option>
              <option
                value="cs"
              >
                Czech 
              </option>
              <option
                value="da"
              >
                Danish 
              </option>
              <option
                value="div"
              >
                Dhivehi 
              </option>
              <option
                value="nl"
              >
                Dutch 
              </option>
              <option
                value="en"
              >
                English 
              </option>
              <option
                value="et"
              >
                Estonian 
              </option>
              <option
                value="fo"
              >
                Faroese 
              </option>
              <option
                value="fa"
              >
                Farsi 
              </option>
              <option
                value="fi"
              >
                Finnish 
              </option>
              <option
                value="fr"
              >
                French 
              </option>
              <option
                value="gl"
              >
                Galician 
              </option>
              <option
                value="ka"
              >
                Georgian 
              </option>
              <option
                value="de"
              >
                German 
              </option>
              <option
                value="el"
              >
                Greek 
              </option>
              <option
                value="gu"
              >
                Gujarati 
              </option>
              <option
                value="he"
              >
                Hebrew 
              </option>
              <option
                value="hi"
              >
                Hindi 
              </option>
              <option
                value="hu"
              >
                Hungarian 
              </option>
              <option
                value="is"
              >
                Icelandic 
              </option>
              <option
                value="id"
              >
                Indonesian 
              </option>
              <option
                value="it"
              >
                Italian 
              </option>
              <option
                value="ja"
              >
                Japanese 
              </option>
              <option
                value="kn"
              >
                Kannada 
              </option>
              <option
                value="kk"
              >
                Kazakh 
              </option>
              <option
                value="kok"
              >
                Konkani 
              </option>
              <option
                value="ko"
              >
                Korean 
              </option>
              <option
                value="ky"
              >
                Kyrgyz 
              </option>
              <option
                value="lv"
              >
                Latvian 
              </option>
              <option
                value="lt"
              >
                Lithuanian 
              </option>
              <option
                value="mk"
              >
                Macedonian
              </option>
              <option
                value="ms"
              >
                Malay 
              </option>
              <option
                value="mr"
              >
                Marathi 
              </option>
              <option
                value="mn"
              >
                Mongolian 
              </option>
              <option
                value="nb"
              >
                Norwegian (Bokmål) 
              </option>
              <option
                value="nn"
              >
                Norwegian (Nynorsk) 
              </option>
              <option
                value="pl"
              >
                Polish 
              </option>
              <option
                value="pt"
              >
                Portuguese 
              </option>
              <option
                value="pa"
              >
                Punjabi 
              </option>
              <option
                value="ro"
              >
                Romanian 
              </option>
              <option
                value="ru"
              >
                Russian 
              </option>
              <option
                value="sa"
              >
                Sanskrit 
              </option>
              <option
                value="sk"
              >
                Slovak 
              </option>
              <option
                value="sl"
              >
                Slovenian 
              </option>
              <option
                value="es"
              >
                Spanish 
              </option>
              <option
                value="sw"
              >
                Swahili 
              </option>
              <option
                value="sv"
              >
                Swedish 
              </option>
              <option
                value="syr"
              >
                Syriac 
              </option>
              <option
                value="ta"
              >
                Tamil 
              </option>
              <option
                value="tt"
              >
                Tatar 
              </option>
              <option
                value="te"
              >
                Telugu 
              </option>
              <option
                value="th"
              >
                Thai 
              </option>
              <option
                value="tr"
              >
                Turkish 
              </option>
              <option
                value="uk"
              >
                Ukrainian 
              </option>
              <option
                value="ur"
              >
                Urdu 
              </option>
              <option
                value="vi"
              >
                Vietnamese 
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Language - Regional Subtag
            </label>
            <div
              class="usa-helptext disabled"
            />
            
            <select
              class="usa-select disabled "
              disabled=""
              name="languageRegSubTag"
              type="select"
            >
              <option
                value=""
              >
                -Select-
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Dataset's IT Unique Investment Identifier
              <div
                class="tooltip"
              >
                <svg
                  height="20px"
                  style="margin-left: .5em;"
                  tabindex="0"
                  width="20px"
                >
                  info.svg
                </svg>
              </div>
            </label>
            <div
              class="usa-helptext "
            />
            
            <input
              class="usa-input  undefined undefined"
              name="primary_it_investment_uii"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Related Documents
              <div
                class="tooltip"
              >
                <svg
                  height="20px"
                  style="margin-left: .5em;"
                  tabindex="0"
                  width="20px"
                >
                  info.svg
                </svg>
              </div>
            </label>
            <div
              class="usa-helptext "
            />
            
            <input
              class="usa-input  undefined undefined"
              name="related_documents"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Release Date
            </label>
            <div
              class="usa-helptext "
            >
              Date Format: MM/DD/YYYY
            </div>
            
            <div>
              <input
                class="usa-input "
                name="release_date"
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              System of Records Notice URL
              <div
                class="tooltip"
              >
                <svg
                  height="20px"
                  style="margin-left: .5em;"
                  tabindex="0"
                  width="20px"
                >
                  info.svg
                </svg>
              </div>
            </label>
            <div
              class="usa-helptext "
            />
            
            <input
              class="usa-input  undefined undefined"
              name="system_of_records"
              type="string"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid-row margin-top-3"
      >
        <div
          class="grid-col-12"
        >
          <div
            class="wrapper"
          >
            <label
              class="usa-label"
            >
              Is parent dataset
            </label>
            <div
              class="usa-helptext "
            />
            
            <select
              choices="Yes,No"
              class="usa-select  "
              name="isParent"
              type="select"
            >
              <option
                value=""
              >
                -Select-
              </option>
              <option
                value="Yes"
              >
                Yes
              </option>
              <option
                value="No"
              >
                No
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="margin-top-6 clearfix"
      >
        <button
          class="usa-button usa-button--outline"
          type="button"
        >
          Save draft
        </button>
        <div
          class="float-right"
        >
          <button
            class="usa-button usa-button--outline"
            type="button"
          >
            Back to previous page
          </button>
          <button
            class="usa-button margin-right-0"
            type="button"
          >
            Save and Continue
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
